lumispy/io_plugins/attolight.py

```diff
@@ -216,6 +216,29 @@ def _calibrate_signal_axis_wavelength(cl_object):
 
         return cl_object
 
+    def _calibrate_signal_axis_wavelength_meanspec(filename, cl_object):
+        ms_file = "HYPCard - Mean spectrum.txt"  # how attomap software names mean spectrum file
+        file = os.path.join(filename, ms_file)  # filename is str; the absolute folder path where the ms_file_name exists
+
+        binning = cl_object.Acquisition_instrument.CCD.binning
+        WL = []  # create empty wavelength array
+
+        # Read in wavelengths from Mean spectrum file, which are in the first column.
+        with open(file, 'r') as f:
+            next(f)
+            for line in f:
+                sep = line.split()
+                WL.append(float(sep[0]) * 1e6)  # wavelength given in mm in file. Convert to nm.
```

Comment:
Also, maybe using the `numpy.loadtxt()` function here is faster. Check how to [here](https://numpy.org/doc/stable/reference/generated/numpy.loadtxt.html?highlight=loadtxt#numpy.loadtxt).